### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by Wes-Isaac**
# Recipe app

In this pull request we implemented the following features:

- A login page built with Devise.
- A registration page built with Devise.
- A food list (with all CRUD implementation, except for 'update').
    - Should display a list of food as in the wireframe.
    - Should lead to the form that allows users to add new food.
- A recipes list (with all CRUD implementation, except for 'update').
     -  Display a list of recipes created by the logged-in user as in the wireframe.
    - Lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- A general shopping list view (all missing food for all your recipes and total price).
     - Show the list of food that is missing for all recipes of the logged-in user.
    -  Sount the total food items and total price of the missing food.
- A list of all public recipes from other users with their names and total prices.
    - Display a list of all public recipes ordered by newest as in the wireframe.
    - Lead to recipe details.

### CODE REVIEW

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class FoodsController < ApplicationController
+  def index
+    @food = current_user.foods.all
+  end
```
Comment (andersonlebon) [inline app/controllers/foods_controller.rb]: - [ ] Kindly note that this should not only display food for the current user rather it should display all foods from the store. 
```.rb
  def index
    @foods = Food.all
  end
```
With this logic, the authorization for deleting the food is ready to be implemented in the view `food#index`.

File: app/views/foods/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ <th><h3>Measurement unit</h3></th>
+ <th><h3>Unit price</h3></th>
+ <th><h3>Actions</h3></th>
+</tr>
+<% @food.each do |food|%>
+<tr> 
+	<td><h3><%= food.name%></h3></td>
+	<td><h3><%= food.measurement_unit%></h3></td>
+	<td><h3><%= food.price%></h3></td>
+	<td><%= button_to "Delete", food_path(food.id), method: :delete, class: "submit" %></td>
```
Comment (andersonlebon) [inline app/views/foods/index.html.erb]: - [ ] Now as you have gotten all foods in the controller kindly allow the current user to delete only his foods.

```html.erb
<td>
  <% if can? :distroy, food %>
     <%= button_to "Delete", food_path(food.id), method: :delete, class: "submit" %>
  <% else %>
    cannot delete this
 <% end %>
</td>

```

File: app/views/foods/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	</tr>
+	<%end%>
+	</table>
+</div>
```
Comment (andersonlebon) [inline app/views/foods/index.html.erb]: ## [OPTIONAL]
- [ ] I would really suggest you frequently use partials to minimize your code.
 example :point_dowmn

```html.erb
<% @foods.each do |food|%>
         <%= render "food", food  %> 
<% end %>
```

File: app/controllers/recipies_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+class RecipiesController < ApplicationController
+  def index
+    @recipes = current_user.recipes
+  end
```
Comment (andersonlebon) [inline app/controllers/recipies_controller.rb]: - [ ] Kindly note that this should not only display Recipes for the current user rather it should display all recipes from the store.

```.rb
  def index
   @recipes= Recipe.all
  end
```
With this logic, the authorization for deleting the Recipe is ready to be implemented in the view `recipe#index`.

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Wes-Isaac  and @mckent05 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions

- [ ] I would also strongly recommend you to improve the design :clap: 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
Hi **@Team,** 
# Congratulations! 🎉 
Your project is complete!There is nothing else to say other than... it's time to merge it :shipit: 

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**